powersimdata/input/grid.py

```diff
@@ -138,3 +152,14 @@ def _univ_eq(ref, test, failure_flag=None):
             print(f"non-matching entries: {', '.join(sorted(nonmatching_entries))}")
             return False
         return True
+
+
+def _get_grid_model_from_scenario_list(source):
+    """Get grid model for a scenario listed in the scenario list
+
+    :param str source: path to MAT-file enclosing the grid data.
+    :return: (*str*) -- the grid model.
+    """
+    scenario_number = int(os.path.basename(source).split("_")[0])
+    slm = ScenarioListManager(Context.get_data_access())
+    return slm.get_scenario_table().loc[scenario_number, "grid_model"]
```

Comment:
You can also use `slm.get_scenario_by_id(scenario_number)`

Reply:
```
>>> from powersimdata.scenario.scenario import Scenario
>>> s = Scenario(1712)
Failed to download ScenarioList.csv from server
Falling back to local cache...
Failed to download ExecuteList.csv from server
Falling back to local cache...
SCENARIO: test | new_bus

--> State
analyze
--> Loading grid
Failed to download ScenarioList.csv from server
Falling back to local cache...
Traceback (most recent call last):
  File "/Users/brdo/CEM/PowerSimData/powersimdata/input/grid.py", line 33, in __init__
    self.model_immutables = ModelImmutables(source)
  File "/Users/brdo/CEM/PowerSimData/powersimdata/network/model.py", line 11, in __init__
    self._check_model(model)
  File "/Users/brdo/CEM/PowerSimData/powersimdata/network/model.py", line 31, in _check_model
    raise ValueError("model must be one of %s" % " | ".join(possible))
ValueError: model must be one of usa_tamu

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/brdo/CEM/PowerSimData/powersimdata/scenario/scenario.py", line 40, in __init__
    self.state = Analyze(self)
  File "/Users/brdo/CEM/PowerSimData/powersimdata/scenario/analyze.py", line 39, in __init__
    self._set_ct_and_grid()
  File "/Users/brdo/CEM/PowerSimData/powersimdata/scenario/analyze.py", line 50, in _set_ct_and_grid
    self.grid = Grid(
  File "/Users/brdo/CEM/PowerSimData/powersimdata/input/grid.py", line 36, in __init__
    _get_grid_model_from_scenario_list(source)
  File "/Users/brdo/CEM/PowerSimData/powersimdata/input/grid.py", line 165, in _get_grid_model_from_scenario_list
    return slm.get_scenario(scenario_number)["grid_model"]
KeyError: 'grid_model'
```
Somehow the Traceback is smaller with this implementation. Not sure why ....